manifests/init.pp

```diff
@@ -239,7 +239,7 @@
     $jenkins_package_class = 'jenkins::direct_download'
   } else {
     $jenkins_package_class = 'jenkins::package'
-    if $repo {
+     if ($repo) and ($::osfamily != 'Windows') {
```

Comment:
It would be better to set the default for `repo` in `params.pp` to `false` for windows rather than insert platform conditional logic.